src/ra.hrl

```diff
@@ -39,17 +39,33 @@
 %% after node restart). Pids are not stable in this sense.
 -type ra_server_id() :: {Name :: atom(), Node :: node()}.
 
+%% Specifies server configuration for a new cluster member.
+%% Subset of  ra_server:ra_server_config().
+%% Both `ra:add_member` and `ra:start_server` must be called with the same values.
+-type ra_new_server() :: #{id := ra_server_id(),
+
+                           %% If set, server will start as non-voter until later promoted by the
+                           %% leader.
+                           non_voter => boolean(),
```

Comment:
I'm not a big fan of booleans I have to admit and I think we should consider how we might want to configure a new member which isn't automatically promoted to a full voter. In this case we have 3 joining member types: `voter | promotable_non_voter | permanent_non_voter`. 

So perhaps something like `membership` => `voter | promotable | non_voter`